test/fuzz/basic.ml

```diff
@@ -0,0 +1,430 @@
+    ; const Ceil
+    ; const Floor
+    ; const Trunc
+    ; const Nearest
+    ]
+
+let fbinop =
+  choose
+    [ const Add
+    ; const Sub
+    ; const Mul
+    ; const Div
+    ; const Min
+    ; const Max
+    ; const Copysign
+    ]
+
+let frelop =
+  choose [ const Eq; const Ne; const Lt; const Gt; const Le; const Ge ]
+
+let fbinop_32 =
+  let+ fbinop in
+  F_binop (S32, fbinop)
+
+let fbinop_64 =
+  let+ fbinop in
+  F_binop (S64, fbinop)
+
+let funop_32 =
+  let+ funop in
+  F_unop (S32, funop)
+
+let funop_64 =
+  let+ funop in
+  F_unop (S64, funop)
+
+let frelop_32 =
+  let+ frelop in
+  F_relop (S32, frelop)
+
+let frelop_64 =
+  let+ frelop in
+  F_relop (S64, frelop)
+
+let const_f32 =
+  let+ float in
+  F32_const (Owi.Float32.of_float float)
+
+let const_f64 =
+  let+ float in
+  F64_const (Owi.Float64.of_float float)
+
+let f32_convert_i32 =
+  let+ sx in
+  F_convert_i (S32, S32, sx)
+
+let f32_convert_i64 =
+  let+ sx in
+  F_convert_i (S32, S64, sx)
+
+let f64_convert_i32 =
+  let+ sx in
+  F_convert_i (S64, S32, sx)
+
+let f64_convert_i64 =
+  let+ sx in
+  F_convert_i (S64, S64, sx)
+
+let i32_trunc_f32 =
+  let+ sx in
+  I_trunc_f (S32, S32, sx)
+
+let i32_trunc_f64 =
+  let+ sx in
+  I_trunc_f (S32, S64, sx)
+
+let i64_trunc_f32 =
+  let+ sx in
+  I_trunc_f (S64, S32, sx)
+
+let i64_trunc_f64 =
+  let+ sx in
+  I_trunc_f (S64, S64, sx)
+
+let i32_trunc_sat_f32 =
+  let+ sx in
+  I_trunc_sat_f (S32, S32, sx)
+
+let i32_trunc_sat_f64 =
+  let+ sx in
+  I_trunc_sat_f (S32, S64, sx)
+
+let i64_trunc_sat_f32 =
+  let+ sx in
+  I_trunc_sat_f (S64, S32, sx)
+
+let i64_trunc_sat_f64 =
+  let+ sx in
+  I_trunc_sat_f (S64, S64, sx)
+
+let f32_demote_f64 = const F32_demote_f64
+
+let f64_promote_f32 = const F64_promote_f32
+
+let i32_reinterpret_f32 = const (I_reinterpret_f (S32, S32))
+
+let i64_reinterpret_f64 = const (I_reinterpret_f (S64, S64))
+
+let f32_reinterpret_i32 = const (F_reinterpret_i (S32, S32))
+
+let f64_reinterpret_i64 = const (F_reinterpret_i (S64, S64))
+
+let global ntyp env =
+  let globals = Env.get_globals ntyp env ~only_mut:false in
+  List.map
+    ( fun (name, (_, _)) ->
+      pair (const (Global_get (Symbolic name))) (const [ S.Push (Num_type ntyp) ]) )
+    globals
+
+let global_i32 = global I32
+
+let global_i64 = global I64
+
+let global_f32 = global F32
+
+let global_f64 = global F64
+
+let global_set ntyp env =
+  let globals = Env.get_globals ntyp env ~only_mut:true in
+  List.map
+    ( fun (name, (_, _)) ->
+      pair (const (Global_set (Symbolic name))) (const [ S.Pop ]) )
+    globals
+
+let global_set_i32 = global_set I32
+
+let global_set_i64 = global_set I64
+
+let global_set_f32 = global_set F32
+
+let global_set_f64 = global_set F64
+
+let local ntyp env =
+  let locals = Env.get_locals ntyp env in
+  List.map
+    ( fun (name, _) -> match name with
+      | Some n -> pair (const (Local_get (Symbolic n))) (const [ S.Push (Num_type ntyp) ])
+      | None -> assert false )
```

Comment:
If this can not happen, then maybe we should change the type of locals in the environment ?

Reply:
It's about `None -> assert false` ?
The reason is that locales are always generated with a symbolic name

Reply:
Yes. I believe we should change the type of locals to be `(string * typ) list` or something similar instead of `param list` to avoid having to handle the `None` case everywhere.

Reply:
Yes. I believe we should change the type of locals in the environment to be `(string * typ) list` or something similar instead of `param list` to avoid having to handle the `None` case everywhere.

Reply:
of course, I didn't understand ;)